src/accounts/models.py

```diff
@@ -83,19 +84,47 @@ class User(AbstractBaseUser):
 
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
 
+    username = models.CharField(
+        _('username'),
+        max_length=30,
+        unique=True,
+        help_text=_('Required. 30 characters or fewer. Letters, digits and @/./+/-/_ only.'),
+        validators=[
+            validators.RegexValidator(
+                r'^[\w.@+-]+$',
+                _('Enter a valid username. This value may contain only letters, numbers and @/./+/-/_ characters.'),
+                'invalid'
+            ),
+        ],
+        error_messages={
+            'unique': _("A user with that username already exists."),
+        }
+    )
+
     first_name = models.CharField(_('First Name'), max_length=50)
+
     last_name = models.CharField(_('Last Name'), max_length=50)
+
     email = models.EmailField(_('Email address'), unique=True)
 
     gender = models.CharField(max_length=1, choices=GENDER_CHOICES, default=GENDER_MALE)
 
     confirmed_email = models.BooleanField(default=False)
 
-    is_staff = models.BooleanField(_('staff status'), default=False)
-    is_superuser = models.BooleanField(_('superuser status'), default=False)
-    is_active = models.BooleanField(_('active'), default=True)
+    is_staff = models.BooleanField(
+        _('staff status'),
+        default=False,
+        help_text=_('Designates whether the user can log into this admin site.')
+    )
+
+    is_active = models.BooleanField(
+        _('active'),
+        default=True,
+        help_text=_('Designates whether this user should be treated as active. '
+                    'Unselect this instead of deleting accounts.')
+    )
 
-    date_joined = models.DateTimeField(_('date joined'), auto_now_add=True)
+    date_joined = models.DateTimeField(_('date joined'), default=timezone.now)
```

Comment:
Probably because you are adding the possibility of changing the date_joined in the form. If you do that, auto_now_add cannot be used. However in this case auto_now_add disables the possibility of you changing the date_joined value, which  for me is safer.